subsys/shell/modules/device_service.c

```diff
@@ -192,10 +193,48 @@ static int cmd_device_list(const struct shell *sh,
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE_RUNTIME
+static int cmd_device_pm_toggle(const struct shell *sh,
+			 size_t argc, char **argv)
+{
+	const struct device *dev;
+	enum pm_device_state pm_state;
+
```

Comment:
IMO to complicated ifdefiry, why not just

```c

	if (!IS_ENABLED(CONFIG_PM_DEVICE_RUNTIME)) {
		return -ENOTSUP;
	}

	dev = device_get_binding(argv[1]);
	if (dev == NULL) {
		shell_error(sh, "Device unknown (%s)", argv[1]);
		return -ENODEV;
	}
```

Reply:
The pm_toggle shell command is completely when the Kconfig is not enabled, happy to leave it around but do the above, there would be a small rom cost add regardless of pm kconfig options then, probably negligible. I’ll wait for others to chime in on this one, I have no strong preference.

Reply:
Flash cost is overrated :-)
You could also use `#if defined(CONFIG_PM_DEVICE_RUNTIME)` inside SHELL_STATIC_SUBCMD_SET_CREATE(),
BT shell does this very extensively, https://github.com/zephyrproject-rtos/zephyr/blob/a7c011b0bee4ca3bd72f192641bee56526a07e7a/subsys/bluetooth/shell/bt.c#L3704-L3714

Reply:
I think I've sorted this out a bit better, the shell cmd is now a define set or unset in the ifdef with the toggle command function, it feels cleaner anyways